c-api/compat-5.3.c

```diff
@@ -14,6 +15,61 @@
 /* definitions for Lua 5.1 only */
 #if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM == 501
 
+#ifndef COMPAT53_LUA_FILE_BUFFER_SIZE
+#define COMPAT53_LUA_FILE_BUFFER_SIZE 4096
+#endif // Lua File Buffer Size
+
+#define compat53_f_parser_buffer COMPAT53_CONCAT(COMPAT53_PREFIX, compat53_f_parser_buffer)
+struct compat53_f_parser_buffer {
+  FILE* file;
+  size_t start;
+  char buffer[COMPAT53_LUA_FILE_BUFFER_SIZE];
+};
+
+#if defined(__GNUC__) && (__GNUC__ < 6)  && defined(__MINGW32__)
+// MinGW 5.x.x and below
+// So as #485 indicates, MinGW doesn't have this function of 5.3 of MinGW
```

Comment:
#485 of what?

Reply:
probably https://github.com/ThePhD/sol2/issues/485, should be added to the source comment

Reply:
I clarified the source.